test_conformance/printf/util_printf.cpp

```diff
@@ -1743,10 +1743,12 @@ size_t verifyOutputBuffer(char *analysisBuffer,testCase* pTestCase,size_t testId
     else if (pTestCase->_correctBuffer[testId] == "INF")
         return strcmp(analysisBuffer, "INF")
             && strcmp(analysisBuffer, "INFINITY");
-    else if (pTestCase->_correctBuffer[testId] == "nan")
-        return strcmp(analysisBuffer, "nan") && strcmp(analysisBuffer, "-nan");
-    else if (pTestCase->_correctBuffer[testId] == "NAN")
-        return strcmp(analysisBuffer, "NAN") && strcmp(analysisBuffer, "-NAN");
+    else if (pTestCase->_correctBuffer[testId] == "nan"
+             || pTestCase->_correctBuffer[testId] == "NAN")
+    {
+        std::regex nanRegex(R"(-?nan(\(.*\))?)", std::regex_constants::icase);
```

Comment:
Looks like this will accept  Nan, nAn, nAN, ... , however I think the appropriate regex is:

std::regex nanRegex(R"(-?)" + pTestCase->_correctBuffer[testId] + R"((\(.*\))?)"); 

Could you check, please?

Reply:
I've modified with your regex but added backslashes in last part  R"((\(.*\))?)"); I've verified it works.

Reply:
Almost there but think std::regex_constants::icase has to be tossed because it makes the regex case insensitive.